### PR TITLE
feat: add ability to attach dates to gifts

### LINKED ISSUE #4710: Add date field for Gifts
As it stands, gifts are recorded in the order they’re entered.  It would be better if we could (optionally) record the date the gift was given/received so that we could later sort by that.

### PR DESCRIPTION
**Pull request by sktlmarat**
Done:

- Added form to add the date (CreateGift.vue)
- Date is stored on the database (gift table)

Requires work:

- Display gift date on dashboard

This close #4710

### CODE REVIEW

**Comment by sktlmarat** [comment]
Dear @asbiin,

Can I please get your feedback on this PR. I would really appreciate it.

Thanks!

File: resources/js/components/people/gifts/CreateGift.vue [hunk 1]
```diff
@@ -109,6 +112,18 @@
...
+        <div v-if="displayDate" class="dt dt--fixed pb3 mb3 bb b--gray-monica">
+          <!-- Date -->
+          <form-input
+            :id="'date'"
+            v-model="newGift.date"
+            :input-type="'date'"
+            :class="'dtc pr2'"
+            :title="$t('people.gifts_add_date')"
+            @submit="store"
+          />
```
Comment (asbiin) [inline resources/js/components/people/gifts/CreateGift.vue:124]: For dates we have a specific component:
```suggestion
          <form-date
            :id="'date'"
            v-model="newGift.date"
            :show-calendar-on-focus="true"
            :locale="locale"
            :class="[ dirltr ? 'fl dtc pr2' : 'fr dtc pr2' ]"
            :label="$t('people.gifts_add_date')"
            @submit="store"
          />
```

**Review by asbiin — CHANGES_REQUESTED** [review]
That's great, see my comments.
I think we should also display the date on display mode (in `Gift.vue`)

**Comment by asbiin** [comment]
This close #4710

**Comment by sktlmarat** [comment]
Hi @asbiin!
That's exactly what I meant by "Display gift date on dashboard".
Thank you for your feedback and addition! All changes were committed.

**Review by asbiin — APPROVED** [review]

**Comment by asbiin** [comment]
thank you @sktlmarat !

**Comment by mgrhm** [comment]
This is brilliant -- I just pulled it and I'm able to add dates to gifts I've given.

I'm not sure if I'm missing something, but is there an option to display the dates on the contact’s page?  As it stands, the date is correctly recorded but I can only see it when I click 'Edit'.

Screen Shot 2021-03-21 at 18 41 01

**Pull request merged**